### PR TITLE
196: Add GP Profile to the WP Profile.

### LINKED ISSUE #196: Integrate with WP's user profile.
Currently the GP profile is a stand alone affair, which is fine, but we should include the same settings in the WP user profile as well.

### PR DESCRIPTION
**Pull request by toolstack**
Split the profile template file in to three, one part for the otter html, one part for the inner editing html and the last part for the inner viewing html.

Added filters and functions to display the GP profile in the WP profile page.

Resolves #196.

See #197 for history.

### CODE REVIEW

File: gp-includes/routes/profile.php [hunk 1]
```diff
@@ -16,17 +16,22 @@ public function profile_get() {
 		$this->tmpl( 'profile' );
 	}
 
-	public function profile_post() {
+	public function profile_post( $user_id = null ) {
 		if ( isset( $_POST['submit'] ) ) {
+			if ( (int)$user_id == 0 ) {
```
Comment (ocean90) [inline gp-includes/routes/profile.php]: Why not `null === $user_id`?

File: gp-includes/misc.php [hunk 2]
```diff
@@ -304,7 +300,6 @@ function gp_is_between_exclusive( $value, $start, $end ) {
 	return $value > $start && $value < $end;
 }
 
-
 /**
  * Acts the same as core PHP setcookie() but its arguments are run through the gp_set_cookie filter.
  *
```
Comment (ocean90) [inline gp-includes/misc.php:310]: That shouldn't be part of this PR.

File: gp-templates/profile-view.php [hunk 3]
```diff
@@ -0,0 +1,47 @@
+<?php
```
Comment (ocean90) [inline gp-templates/profile-view.php]: Can you explain the purpose of this file? If it doesn't get used by GlotPress it should probably be part of the plugin.

Comment (toolstack) [inline gp-templates/profile-view.php]: I flip flopped a few times on it but it seemed better to keep in in the core as it's really a "read only" version of the edit file.

Since custom theme's might make changes to the edit file I thought it best that they have the opportunity to change the view file at the same time to retain a consistent look/feel.

File: gp-includes/misc.php (hunk 2 above)
Comment (toolstack) [inline gp-includes/misc.php:310]: I'll pull it out, just noticed it while editing the file.

File: gp-includes/routes/profile.php (hunk 1 above)
Comment (toolstack) [inline gp-includes/routes/profile.php]: In my testing I found sometimes we may get a 0 passed in from WordPress, so the null check would not set the current user id correctly.

Comment (ocean90) [inline gp-includes/routes/profile.php]: It would be nice to know where the 0 comes from. IMO is this a case where a simple `if ( ! $user_id )` would be enough.

Related: https://make.wordpress.org/core/handbook/best-practices/coding-standards/php/#yoda-conditions
=> `if ( 0 == (int) $user_id ) {` or better:

```
$user_id = (int) $user_id;
if ( 0 === $user_id ) { 
…
}
```

**Comment by ocean90** [comment]
@toolstack Added another comment for the $user_id check. Choose one option and merge afterwards. :+1:
